### DISCUSSION TITLE
Can I install Saleor without docker?

### DISCUSSION THREAD

**Discussion by noelalam9999** [saleor/saleor#7940] [Deployment]
Hello, I am newbee in backend development. I have been researching on Saleor and found it to be a very clean system for developing an e-commerce website fast. However I was wondering how I could deploy the system manually without docker-compose. Thanks in advance!

**Comment by karolkielecki** [comment]
Hi! Saleor's preferred deployment method is Docker. But if you've installed node.js, go to the catalog and run `npm install` and `npm start`--that's how you will run it outside docker.

**Reply by amityweb** [reply]
I have npm ERR! Missing script: "start" for npm start. I can add it into package but what is start calling? Thanks

**Reply by zelfroster** [reply]
> I have npm ERR! Missing script: "start" for npm start. I can add it into package but what is start calling? Thanks

Saleor-dashboard has moved from cra to vite now, so you will have to run `npm run dev` now. I had run into same issue.

**Comment by Diego17ccs** [comment]
sorry if I'm late to answer, But the answer to your question is YES, the installation process does not consist only of npm install and npm start, the installation process as such has enough details that for reasons that are really unknown are not in the documentation of the site that they provide, since they do not explain the installation mode without docker, therefore they ignore endless details to install it in local or development mode and not to mention the endless pitfalls to install it in production mode, in most of the cases that I have been able to see before the same or variations of the question you are asking, the answer is that usually you are referred to a link on youtube of how the installation process is done through a person of Hindu nationality who uses a little methodology conventional

**Reply by karolkielecki** [reply]
@Diego17ccs Hi, Saleor's preferred method is Docker, as stated in the documentation. If you have more suggestions on what we should include in our documentation, please let me know. Then I'll pass your feedback to the team and discuss it.

**Comment by noelalam9999** [comment]
Its ok. I understand Hindi. If the video shows how I can deploy a working
Saleor system in production mode that would be fine by me. Could you please
refer me the link?

Regards,
 Noel Alam

**Reply by Diego17ccs** [reply]
The truth is that I have not seen those links on the web or on YouTube for a long time, the last time I was looking for them to pass them to a friend so that he had a reference of certain aspects of the saleor installation process, but in the end I looked for them and I did not find them again, in the same way what if I can tell you if I remember correctly those videos served as a reference but not as an absolute guide, what I can do is redirect you to a computer company called VenSoft, C. A. dedicated to the development of application programs and pages and that in turn is dedicated to offering the installation of saleor where the client selects and under the modality that the client selects, this company gives you the installation manuals under the modality that the client selects, well-detailed manuals, step by step in the process of installing the saleor platform in production mode without docker, manual or guide that covers each and every one of the aspects that the people of saleor for the same reason that i tell you before in previous cases they do not cover, likewise this company that I am mentioning to you, provides assistance throughout the installation and configuration process of the saleor platform for a much cheaper price than the service provided by the people of saleor in the plans that They offer in the cloud monthly, in the end you end up with the saleor platform installed where you have it and with a manual or well-detailed guide and without any type of limitation in the database or in the software, apart if you already requested a installation service of the saleor platform with this company that I mentioned, you can request assistance for what comes after the installation that would be the modification of the storefront, they will redirect you as a developer to fulfill that task of modifying the storefront, which in terms of what the storefron refers to is something that on the part of the people of saleor is not documented in full details for reasons that are really unknown until today ! 

If you are interested, I will leave you the link of the facebook profile of the founder CEO of VenSoft, C. A. so that you can get in touch and solve all your questions, doubts and answers regarding the saleor platform

https://www.facebook.com/diego.soules/?viewas=100000686899395&show_switched_toast=0&show_switched_tooltip=0&show_podcast_settings=0

**Reply by karolkielecki** [reply]
@noelalam9999 Please check the following video tutorial provided by one of our community members: https://www.youtube.com/playlist?list=PLzt7OcHd8ypH9mq6bwttlON3PiUzB-GJl

**Comment by tiendq** [comment]
> Hello, I am newbee in backend development. I have been researching on Saleor and found it to be a very clean system for developing an e-commerce website fast. However I was wondering how I could deploy the system manually without docker-compose. Thanks in advance!

Yes, you can. I did it, it depends on how you want to use it. If you just need to run it locally then it's much less configuration effort than for public access.

**Comment by timuric** [comment]
@noelalam9999 if you don't want to deal with the devops, best way is to run a cloud instance. You can setup your storefront in minutes by running `npx saleor storefront create --demo`

**Reply by liwenz** [reply]
is there a document that install without docker?

**Reply by tiendq** [reply]
I did it successfully a couple of year ago, just look at Docker files and try and surely you will need to fix many issues. But in recent releases it seems harder to run Saleor without Docker so I decided to build my own Docker version since there're many thing in default Docker files which difficult to manage.

**Comment by liwenz** [comment]
https://docs.saleor.io/docs/3.x/setup/windows
Manual Installation on Windows

https://github.com/liwenz/Deploy_Saleor
A bash script for Production Deployment of [Saleor](https://github.com/mirumee/saleor)  linux
